src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1274,5 +1274,35 @@ internal BoundStatement NoOp(NoOpStatementFlavor noOpStatementFlavor)
         {
             return new BoundNoOpStatement(Syntax, noOpStatementFlavor);
         }
+
+        internal BoundLocal MakeTempForDiscard(BoundDiscardedExpression node, ArrayBuilder<LocalSymbol> temps)
+        {
+            LocalSymbol temp;
+            BoundLocal result = MakeTempForDiscard(node, out temp);
+            temps.Add(temp);
+            return result;
+        }
+
+        internal BoundLocal MakeTempForDiscard(BoundDiscardedExpression node, out LocalSymbol temp)
+        {
+            temp = new SynthesizedLocal(this.CurrentMethod, node.Type, SynthesizedLocalKind.LoweringTemp);
+
+            return new BoundLocal(node.Syntax, temp, constantValueOpt: null, type: node.Type) { WasCompilerGenerated = true };
+        }
+
+        internal ImmutableArray<BoundExpression> MakeTempsForDiscardArguments(ImmutableArray<BoundExpression> arguments, ArrayBuilder<LocalSymbol> builder)
+        {
+            var discardsCount = arguments.Count(a => a.Kind == BoundKind.DiscardedExpression);
+
+            if (discardsCount != 0)
+            {
+                arguments = arguments.SelectAsArray(
+                                            arg => arg.Kind == BoundKind.DiscardedExpression ?
+                                                this.MakeTempForDiscard((BoundDiscardedExpression)arg, builder) :
```

Comment:
>this.MakeTempForDiscard((BoundDiscardedExpression)arg, builder) [](start = 48, length = 63)

I think we would prefer to avoid using lambdas that need a closure.